org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/LiveShadowedChunkedContainer.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2017, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.resource.containers;
+
+import static org.eclipse.xtext.xbase.lib.IterableExtensions.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.IContainer;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.resource.IResourceDescription;
+import org.eclipse.xtext.resource.impl.ChunkedResourceDescriptions;
+import org.eclipse.xtext.resource.impl.LiveShadowedChunkedResourceDescriptions;
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsData;
+import org.eclipse.xtext.workspace.IProjectConfig;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+
+import com.google.common.annotations.Beta;
+import com.google.common.collect.Iterables;
+
+/**
+ * @author koehnlein - Initial contribution and API
+ * @since 2.14
+ */
+@Beta
+public class LiveShadowedChunkedContainer implements IContainer {
+	private final LiveShadowedChunkedResourceDescriptions descriptions;
+
+	private final String containerName;
+
+	private IProjectConfig projectConfig;
+
+	private boolean isProjectConfigSet = false;
+
+	public LiveShadowedChunkedContainer(LiveShadowedChunkedResourceDescriptions descriptions, String containerName) {
+		this.descriptions = descriptions;
+		this.containerName = containerName;
+	}
+
+	protected ChunkedResourceDescriptions getChunkedResourceDescriptions() {
+		return (ChunkedResourceDescriptions) descriptions.getGlobalDescriptions();
+	}
+
+	protected IProjectConfig getProjectConfig() {
+		if (!isProjectConfigSet) {
+			projectConfig = descriptions.getWorkspaceConfig() == null ? null
+					: descriptions.getWorkspaceConfig().findProjectByName(this.containerName);
```

Comment:
Unneeded `this`, this can be fixed using cleanup actions normally

Reply:
Don't like that my code gets modified when I save. Usually have them disabled. But thx for the hint.